torch/export/exported_program.py

```diff
@@ -206,21 +206,14 @@ def assert_valid_to_preserve(op_overload):
                     f"{op_overload} is a TorchScript op, we can't preserve it as is"
                 )
 
-            if not torch._C._dispatch_has_kernel_for_dispatch_key(
```

Comment:
hmmm when @tugsbayasgalan first added it, the argument was that the way the dispatcher is set up today, the only reason we really need this `_preserve_ops` list in the first place is because of CIA ops, so we may as well make that explicit and loudly tell the user when they are adding ops into the `_preserve_ops` list unnecessarily (if you put a non-CIA op in that list, we already won't decompose it. And if you include something in both preserve_ops and in your decomp table, one could argue that's a user error and you should just rationalize those two arguments so they don't overlap).

Although it looks like the extra benefit from this change would be that a user can more easily express e.g. "run all core aten ops, **but** preserve [some_op_that_core_aten_normally_decomposes,_but_is_not_CIA]". Is that fair? (that seems like a reasonable UX improvement, since I guess not everyone will specify a decomp table, and some people will just default to using core aten decomps)

Reply:
Yeah, that was what I was trying to change the behavior to. Users most likely just want to prevent decompositions on a specific operator, and don't know what operator is CIA or not -- also not sure if it's a good idea to expose the idea of CIA since it's an implementation detail too.